### PR TITLE
Update docs/optlib

### PR DESCRIPTION
**Pull request by dreamtigers**
* Fixed typos
* Rewrote some paragraphs for clarity
* Reviewed previously unreviewed sections (specially the ones of `_roles`, `_fields`, `_extras`)

In lines 1573 to 1575 and 1621 to 1626 and maybe others, I used the first person ("Here *I* will explain...") because the paragraphs were written that way, but in no way that means my person (@dreamtigers) is the voice doing the explanation. In any case, it should be @masatake, with their permission. If it's an issue, I can rewrite those sections to use a neutral voice.

Before merging this, I will open an issue to discuss some doubts I have with some sections. Should those doubts be resolved, I can commit the additional changes, and then this pull request can be merged successfully.

### CODE REVIEW

**Comment by masatake** [comment]
> Before merging this, I will open an issue to discuss some doubts I have with some sections. Should those doubts be resolved, I can commit the additional changes, and then this pull request can be merged successfully.

I see. THANK YOU VERY MUCH.

**Comment by masatake** [comment]
Neutral voice is better. I'm happy if you rewrite the sentences in neutral voice.
Not only the document is improved, I can learn English writing from your change.

If you are interested in improving the other parts of ctags in long term, let me know.
I will invite you as a member of this project. So you can merge your changes quickly.

I cannot do much for you in rerun but I can help you to convert Elixir parser to mtable-regex based :-P.

**Comment by dreamtigers** [comment]
I will resume work with this in a few days.

Thank you for the compliment! Although my English isn't so good, hope it serves you well!

Thank you for the invitation! But I have to refuse right now due to internet availability issues. When that improves, I'll let you know!

**Comment by masatake** [comment]
I see.

File: docs/optlib.rst [hunk 1]
```diff
@@ -118,11 +124,11 @@ defining a parser, and have extended existing options. Defining
 a new parser with the options is more than "customizing" in
 Universal-ctags.
 
-To make it easier to maintain a parser defined using the options, you can put
-each parser language in a different options file. Universal-ctags doesn't
-preload a single file. Instead, Universal-ctags loads all files having the
-:file:`.ctags` extension under the previously specified directories. If you have
-multiple parser definitions, put them in different files.
+To make easier the maintenance a parser defined using the options, you can put
+each language parser in a different options file. Universal-ctags doesn't
```
Comment (masatake) [inline docs/optlib.rst]: SELF NOTE: each language parser => each parser

**Comment by masatake** [comment]
Though using natural voice is better, the changes improve the documents much. So I will merge this after modifying the change a bit.

**Pull request merged**